dwave/system/temperatures.py

```diff
@@ -249,373 +384,685 @@ def maximum_pseudolikelihood_temperature(bqm=None,
 
         https://www.jstor.org/stable/25464568
 
-    '''
-
-    T_estimate = 0
-    T_bootstrap_estimates = np.zeros(num_bootstrap_samples)
-
-    #Check for largest local excitation in every sample, and over all samples
-    if site_energy is None:
-        if bqm is None or sampleset is None:
-            raise ValueError('site_energy can only be derived if both'
-                             'bqm and sampleset are provided as arguments')
-        site_energy = effective_field(bqm,
-                                      sampleset,
-                                      current_state_energy = True)
-    max_excitation = np.max(site_energy[0],axis=1)
-    max_excitation_all =  np.max(max_excitation)
-    if max_excitation_all <= 0:
-        #There are no local excitations present in the sample set, therefore
-        #the temperature is estimated as 0. 
-        pass
+    """
+    x0 = None
+    bisect_bracket = None
+    if T_guess is not None:
+        x0 = -1 / T_guess
+
+    if optimize_method == "bisect":
+        if not 0 <= T_bracket[0] < T_bracket[1]:
+            raise ValueError("Bad T_bracket, must be positive ordered scalars.")
+        bisect_bracket = [-1 / T_bracket[i] for i in range(2)]
+        if x0 is not None and (x0 < bisect_bracket[0] or x0 > bisect_bracket[1]):
+            raise ValueError(f"x0 {x0} and T_bracket {T_bracket} are inconsistent")
+    if x0 is not None:
+        x0 = [x0]
+    x, x_bs = maximum_pseudolikelihood(
+        bqms=[bqm],
+        sampleset=sampleset,
+        en1=en1,
+        num_bootstrap_samples=num_bootstrap_samples,
+        seed=seed,
+        x0=x0,
+        optimize_method=optimize_method,
+        bisect_bracket=bisect_bracket,
+        sample_weights=sample_weights,
+    )
+    # exp(-1/Teff Hp) = exp(estimator[0] Hp)
+    return -1 / x, -1 / x_bs
+
+
+def maximum_pseudolikelihood(
```

Comment:
This has become a very long function with many branches. Could it be split up and organized into smaller functions for the various cases and checks? That would make it much easier to review and maintain, and would also allow for better documentation and comments.